src/wp-includes/class-wp-script-modules.php

```diff
@@ -225,6 +226,32 @@ public function print_import_map() {
 		}
 	}
 
+	/**
+	 * Prints the necessary script to load import map polyfill for browsers that
+	 * do not support import maps.
```

Comment:
It could also be clarified that the script gets printed only when there is an import map to print.

By the way, have you considered printing it next to the import map instead as part of `print_import_map`?

Reply:
There are no special benefits on doing it, and would force to add the polyfill to the `head` in all block themes.
Also, if we deprecate in a future old browsers, we can deprecate the function.

Reply:
If we move the logic to printing the polyfill helper then we won’t ever need to deprecate any method, and there will no no need to register another action. 

I guess it can remain as is, but it isn't that obvious one is better than another. There are no measurements for the performance impact so something I wouldn't take for granted as the polyfill needs to load only for 10-ish % of users and it actually needs to load for them before the entries listed in the import map get imported. 

Finally, in the [usage section of the polyfill](https://github.com/guybedford/es-module-shims?tab=readme-ov-file#usage) it is printed just before the import map so this should be carefully tested.

Reply:
Let's move it then, also before the import map. I will make another testing battery in browserstack asap.